File: Programmers Level1/3진법뒤집기.cpp
#include <string>
#include <vector>
#include <math.h>
using namespace std;

int solution(int n) {
	int answer = 0;
	vector<int> a; // 45를 3으로 나눠서 나머지값을 넣는다.
	while(n>=3){
	a.push_back(n % 3);
	n = n / 3;
	if (n < 3) {
		a.push_back(n);
		break;
	}
	}
	int sum = 0;
	for (int i = a.size()-1; i >= 0; i--) {
		answer += (pow(3, sum )*a[i]);
		sum++;
	}
	return answer;
}
int main() {
	int n = 45;
		solution(n);
}

File: Programmers Level1 - Kotlin/3진법 뒤집기.kt
import kotlin.math.pow

fun main() {
    solution(125)
}
fun solution(n: Int): Int {
    var number = n
    var answer: Int = 0
    var stringAnswer = ""
    while(number>=3) {// 45
        stringAnswer += number%3
        number /= 3
    }
    stringAnswer += number
    var array = ArrayList<Int>()
    for(i in stringAnswer) {
        array.add(i.toInt()-48)
    }

    for((po, i) in (array.size-1 downTo 0).withIndex()) {
        answer += array[i]* 3.0.pow(po).toInt()
    }

    return answer
}